### PR TITLE
Adds `denoising_end` parameter to ControlNetPipeline for SDXL

### LINKED ISSUE #6159: StableDiffusionXLControlNetXSPipeline does not support  keyword argument 'denoising_end'
### Describe the bug

StableDiffusionXLControlNetXSPipeline does not support  'denoising_end' for use in combination with the SDXL refiner pipeline.

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
[<ipython-input-1-68f36004df4a>](https://5ot3l8cauyj-496ff2e9c6d22116-0-colab.googleusercontent.com/outputframe.html?vrz=colab_20231211-060141_RC00_589772369#) in <cell line: 35>()
     33 
     34 # generate image
---> 35 image = pipe(
     36     prompt, controlnet_conditioning_scale=controlnet_conditioning_scale, image=canny_image, denoising_end=0.8, output_type="latent"
     37 ).images

[/usr/local/lib/python3.10/dist-packages/torch/utils/_contextlib.py](https://5ot3l8cauyj-496ff2e9c6d22116-0-colab.googleusercontent.com/outputframe.html?vrz=colab_20231211-060141_RC00_589772369#) in decorate_context(*args, **kwargs)
    113     def decorate_context(*args, **kwargs):
    114         with ctx_factory():
--> 115             return func(*args, **kwargs)
    116 
    117     return decorate_context

TypeError: StableDiffusionXLControlNetXSPipeline.__call__() got an unexpected keyword argument 'denoising_end'
```

### Reproduction

```py
#%pip install --quiet --upgrade git+https://github.com/huggingface/diffusers transformers accelerate mediapy opencv-python 

from diffusers import StableDiffusionXLControlNetXSPipeline, ControlNetXSModel, AutoencoderKL
from diffusers.utils import load_image
import numpy as np
import torch

import cv2
from PIL import Image

prompt = "aerial view, a futuristic research complex in a bright foggy jungle, hard lighting"
negative_prompt = "low quality, bad quality, sketches"

# download an image
image = load_image(
    "https://hf.co/datasets/hf-internal-testing/diffusers-images/resolve/main/sd_controlnet/hf-logo.png"
)

# initialize the models and pipeline
controlnet_conditioning_scale = 0.5  # recommended for good generalization
controlnet = ControlNetXSModel.from_pretrained("UmerHA/ConrolNetXS-SDXL-canny", torch_dtype=torch.float16)
vae = AutoencoderKL.from_pretrained("madebyollin/sdxl-vae-fp16-fix", torch_dtype=torch.float16)
pipe = StableDiffusionXLControlNetXSPipeline.from_pretrained(
    "stabilityai/stable-diffusion-xl-base-1.0", controlnet=controlnet, vae=vae, torch_dtype=torch.float16
)
pipe.enable_model_cpu_offload()

# get canny image
image = np.array(image)
image = cv2.Canny(image, 100, 200)
image = image[:, :, None]
image = np.concatenate([image, image, image], axis=2)
canny_image = Image.fromarray(image)

# generate image
image = pipe(
    prompt, controlnet_conditioning_scale=controlnet_conditioning_scale, image=canny_image, denoising_end=0.8, output_type="latent"
).images

image
```

### Logs

_No response_

### System Info

- `diffusers` version: 0.25.0.dev0
- Platform: Linux-5.15.120+-x86_64-with-glibc2.35
- Python version: 3.10.12
- PyTorch version (GPU?): 2.1.0+cu118 (True)
- Huggingface_hub version: 0.19.4
- Transformers version: 4.36.0
- Accelerate version: 0.25.0
- xFormers version: not installed
- Using GPU in script?: Yes
- Using distributed or parallel set-up in script?: No

aka Colab T4 free tier,

### Who can help?

@yiyixuxu @sayakpaul @DN6 @patrickvonplaten @UmerHA

### PR DESCRIPTION
**Pull request by UmerHA**
# What does this PR do?

Adds `denoising_end` parameter to ControlNetPipeline for SDXL.

~Fixes #6159~ Edit: Removed changes to ControlNetXS, so does not fix #6159 yet.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [x] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

Core library:
- Pipelines:  @patrickvonplaten and @sayakpaul

### CODE REVIEW

**Comment by sayakpaul** [comment]
Thank you! Do we need to add any "# Copied from ..." statements?

**Comment by sayakpaul** [comment]
@UmerHA do you want to revive this PR?

**Comment by UmerHA** [comment]
@sayakpaul Yes, I'll do it after I'm done with the new ControlNetXS PR, which should be ~next week.

**Comment by yiyixuxu** [comment]
can we maybe only keep change for Controlnet pipeline in this PR? 
we have https://github.com/huggingface/diffusers/pull/6772 going on for controlnet XS that kinda races with this PR, no?

**Comment by UmerHA** [comment]
@yiyixuxu Sure - will get it done on Monday!

**Comment by UmerHA** [comment]
@yiyixuxu Have removed the ControlNetXS part - ready for review!

**Review by yiyixuxu — APPROVED** [review]
thanks!

**Comment by yiyixuxu** [comment]
also cc @stevhliu here for awareness
not sure if it's worth mentioning on the doc but now we can use controlnet SDXL pipeline together with the refiner

**Review by sayakpaul — COMMENTED** [review]
Thanks for working on this. 

Should we also add a test for this as I imagine `denoising_end` is something the users will want to experiment with. 

@yiyixuxu WDYT?

**Comment by stevhliu** [comment]
> not sure if it's worth mentioning on the doc but now we can use controlnet SDXL pipeline together with the refiner

Sure, let's add a `[!TIP]` to this [doc](https://huggingface.co/docs/diffusers/main/en/using-diffusers/controlnet#controlnet-with-stable-diffusion-xl) saying you can use `StableDiffusionXLControlNetPipeline` with refiner checkpoints.

**Comment by yiyixuxu** [comment]
@UmerHA 
do you want to add a test? 
we can merge soon:)

File: docs/source/en/using-diffusers/controlnet.md [hunk 1]
```diff
@@ -429,6 +429,27 @@ image = pipe(
 make_image_grid([original_image, canny_image, image], rows=1, cols=3)
 ```
 
+<Tip>
+
+You can use a refiner model with `StableDiffusionXLControlNetPipeline` to improve image quality, just like you can with a regular `StableDiffusionXLPipeline`.
+See [section `Refine image quality` on the `StableDiffusionXL` page](https://huggingface.co/docs/diffusers/using-diffusers/sdxl#refine-image-quality) for how to do it.
```
Comment (stevhliu) [inline docs/source/en/using-diffusers/controlnet.md:435]:
```suggestion
See the [Refine image quality](./sdxl#refine-image-quality) section to learn how to use the refiner model.
```

File: docs/source/en/using-diffusers/controlnet.md [hunk 2]
```diff
@@ -429,6 +429,27 @@ image = pipe(
...
+Make sure to use `StableDiffusionXLControlNetPipeline` and pass `image` and `controlnet_conditioning_scale`.
+
+```
```
Comment (stevhliu) [inline docs/source/en/using-diffusers/controlnet.md:438]:
```suggestion
```py
```

**Review by stevhliu — APPROVED** [review]
A few minor nits, but otherwise LGTM!

**Comment by yiyixuxu** [comment]
thank you:)

**Pull request merged**